Build error `numeric_limits` is not a member of `std`

I ran into the following error when trying to build on Ubuntu-21.10

```bash
[ 96%] Building CXX object CMakeFiles/libopenmc.dir/src/external/quartic_solver.cpp.o
/kaggle/working/openmc/src/external/quartic_solver.cpp:12:33: error: ‘numeric_limits’ is not a member of ‘std’
   12 | constexpr double MACHEPS = std::numeric_limits<double>::epsilon();
      |                                 ^~~~~~~~~~~~~~
/kaggle/working/openmc/src/external/quartic_solver.cpp:12:48: error: expected primary-expression before ‘double’
   12 | constexpr double MACHEPS = std::numeric_limits<double>::epsilon();
      |                                                ^~~~~~
```

It seems to be caused by GCC-11 https://www.gnu.org/software/gcc/gcc-11/porting_to.html#header-dep-changes.

> Some C++ Standard Library headers have been changed to no longer include other headers that they do need to depend on. As such, C++ programs that used standard library components without including the right headers will no longer compile.
> 
> The following headers are used less widely in libstdc++ and may need to be included explicitly when compiled with GCC 11:
> 

>    -  <limits> (for std::numeric_limits)
>    -  <memory> (for std::unique_ptr, std::shared_ptr etc.)
>    -  <utility> (for std::pair, std::tuple_size, std::index_sequence etc.)
>    -  <thread> (for members of namespace std::this_thread.)


Adding `#include <limits>` to `quartic_solver.cpp` solves the issue.

--- a/src/external/quartic_solver.cpp
+++ b/src/external/quartic_solver.cpp
@@ -2,6 +2,7 @@
 #include <cmath>
 #include <complex>
 #include <cstdlib>
+#include <limits>
 
 namespace oqs {
 

